runestone/server/componentdb.py

```diff
@@ -66,20 +66,19 @@ def addQuestionToDB(self):
         srcpath, line = self.state_machine.get_source_and_line()
         subchapter = os.path.basename(srcpath).replace('.rst','')
         chapter = srcpath.split(os.path.sep)[-2]
+        grading_type = self.options.get('autograde', None)
+        print (self.arguments[0], grading_type)
 
         sel = select([questions]).where(and_(questions.c.name == self.arguments[0],
                                               questions.c.base_course == basecourse))
         res = engine.execute(sel).first()
         try:
             if res:
-                if res['question'] != self.block_text:
-                    stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed)
-                    engine.execute(stmt)
+                stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed, is_private='F',
+question_type=self.name, subchapter=subchapter, grading_type = grading_type, author=author,difficulty=difficulty,chapter=chapter)
```

Comment:
If the source code for a directive has not changed there is no reason to update the database.